File: P4Analyst/GraphForP4/ViewModels/CalculatedData.cs
﻿using System.Collections.Generic;

namespace GraphForP4.ViewModels
{
    public class CalculatedData
    {
        public IEnumerable<IEnumerable<ViewNode>> ControlFlowGraphs { get; set; }
        public IEnumerable<IEnumerable<ViewNode>> DataFlowGraphs { get; set; }
        public BarChartData ReadAndWriteChartData { get; set; }
        public PieChartData UseVariable { get; set; }
        public PieChartData Useful { get; set; }
        public BarChartData Headers { get; set; }
        public FileData File { get; set; }
    }
}

File: P4Analyst/AngularApp/ClientApp/src/app/models/calculate/calculatedData.ts
import { Node } from '../graph/node';
import { BarChartData } from './barChartData';
import { PieChartData } from './pieChartData';
import { FileData } from '../file';

export interface CalculatedData {
  controlFlowGraphs: Array<Node[]>;
  dataFlowGraphs: Array<Node[]>;
  readAndWriteChartData: BarChartData;
  useVariable: PieChartData;
  useful: PieChartData;
  headers: BarChartData;
  file: FileData;
}
